google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java

```diff
@@ -1426,4 +1438,80 @@ public void onCompleted() {
       return done;
     }
   }
+
+  @Test
+  public void testBuildingBundleWhenDocumentDoesNotExist() throws Exception {
+    FirestoreBundle.Builder bundleBuilder = new FirestoreBundle.Builder("test-bundle");
+    DocumentSnapshot snapshot = randomDoc.get().get();
+    bundleBuilder.add(snapshot);
+
+    // Expected bundle elements are [bundleMetadata, documentMetadata]
+    List<BundleElement> elements = toBundleElements(bundleBuilder.build().toByteBuffer());
+    assertEquals(2, elements.size());
+
+    verifyMetadata(
+        elements.get(0).getMetadata(),
+        snapshot.getReadTime().toProto(),
+        /*totalDocuments*/ 1,
+        /*expectEmptyContent*/ false);
+    assertEquals(
+        BundledDocumentMetadata.newBuilder()
+            .setExists(false)
+            .setName(fullPath(randomDoc, firestore.getOptions()))
+            .setReadTime(snapshot.getReadTime().toProto())
+            .build(),
+        elements.get(1).getDocumentMetadata());
+  }
+
+  @Test
+  public void testBuildingBundleWithLimitAndLimitToLastQueries() throws Exception {
```

Comment:
Can we split this test into two, one for the limit First, and one for limit last? If either case ever fails it'd be nice to tell independently which is failing.

Reply:
My thought was this provides a good contrast and helps reader understand two cases.

I don't feel strong, I can split if you insist.

Reply:
If the code overhead is minimal I'd prefer the two separate tests for clarity of individual scenarios.

Reply:
Done.